File: solutions/c/easy/sum-of-digits-of-a-five-digit-number/solution-approach1.c
// ──────────────────────────────────────────────────
// Link        https://www.hackerrank.com/contests/forloops1/challenges/sum-of-digits-of-a-five-digit-number/problem?isFullScreen=true
// Problem     Sum of Digits of a Five Digit Number
// Difficulty  Easy
// Subdomain   N/A
// Platform    HackerRank
// Language    c
// Status      Accepted
// Submitted   2026-09-21, 10:33 p.m.
// ──────────────────────────────────────────────────

#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int main() {
	
    int n;
    scanf("%d", &n);
    int sum =0;
    //Complete the code to calculate the sum of the five digits on n.
    while(n!=0){
        int r = n%10;
        sum=sum+r;
        n=n/10;
        
    }
    printf("%d",sum);
    return 0;
}

File: solutions/java15/medium/for-loop7/solution-approach1.java
// ──────────────────────────────────────────────────
// Link        https://www.hackerrank.com/contests/forloops1/challenges/for-loop7/problem?isFullScreen=true
// Problem     for loop7
// Difficulty  Medium
// Subdomain   N/A
// Platform    HackerRank
// Language    java15
// Status      Accepted
// Submitted   2026-09-21, 10:17 p.m.
// ──────────────────────────────────────────────────

import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int sum=0;
        for(int i =1;i<n;i++){
            if(n%i==0){
                sum=sum+i;
            }
        }
        if(sum==n){
            System.out.print(n+" is a Perfect Number");
        }else{
            System.out.print(n+" is not a Perfect Number");
        }
            }
}
